File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章笨小猴.java
//笨小猴   lanqiaoOJ 527

import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str = scanner.next();
        char[] c = str.toCharArray();
        int []length = new int[26];
        int max = -1, min = 101;
        for(int i = 0; i < c.length; i++)
            length[c[i] - 'a']++;
        for(int i = 0; i < 26; i++) {
            if(length[i] == 0)
                continue;
            if(length[i] > max)
                max = length[i];
            if(length[i] < min)
                min = length[i];
        }
        if(c.length == max)
            min = 0;
        boolean ans = is_prime(max - min);
        if(ans) {
            System.out.println("Lucky Word");
            System.out.println(max - min);
        }else {
            System.out.println("No Answer");
            System.out.println(0);
        }
    }
    
    static boolean is_prime(int n) {
        if (n < 2)
            return false;
        for (int i = 2; i <= Math.sqrt(n); i++)
            if (n % i == 0)
                return false;
        return true;
    }
}

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章笨小猴.py
#笨小猴   lanqiaoOJ 527

from math import *
def is_prime(n):
    if n == 0 or n == 1: return False
    m = int(sqrt(n)+1)
    for i in range(2,m):
        if n % i == 0:   return False
    return True
s = input()
maxn = -1
minn = 1000
for i in s:
    n = s.count(i)
    maxn = max(maxn,n)
    minn = min(minn,n)
if is_prime(maxn - minn):
    print("Lucky Word");  print(maxn - minn)
else:
    print("No Answer");   print(0)
